### PR TITLE
[rocksdb] Update to v6.6.4

### PR DESCRIPTION
**Pull request by kobykahane**
Update the rocksdb port to v6.6.4.
- Update CMakeLists.txt patches.
- Remove obsolete patch for zlib findpackage.
- Add patch that adds CONFIG to lz4 findpackage (otherwise building the lz4 feature fails).

Compilation tested with x64-windows-static and x86-windows-static.

### CODE REVIEW

File: ports/rocksdb/portfile.cmake [hunk 1]
```diff
@@ -5,15 +5,15 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 vcpkg_from_github(
```
Comment (NancyLi1013) [inline ports/rocksdb/portfile.cmake]: `include(vcpkg_common_functions)` is no longer needed.
Could you please remove it?

File: ports/rocksdb/portfile.cmake [hunk 2]
```diff
@@ -5,15 +5,15 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
     0002-only-build-one-flavor.patch
-    0003-zlib-findpackage.patch
-    0004-use-find-package.patch
-    0005-static-linking-in-linux.patch
+    0003-use-find-package.patch
+    0004-static-linking-in-linux.patch
+    0005-lz4-config.patch
 )
 
 file(REMOVE "${SOURCE_PATH}/cmake/modules/Findzlib.cmake")
```
Comment (NancyLi1013) [inline ports/rocksdb/portfile.cmake]: Could you please update 
```
file(INSTALL ${SOURCE_PATH}/LICENSE.Apache DESTINATION ${CURRENT_PACKAGES_DIR}/share/rocksdb RENAME copyright)
file(COPY ${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake ${SOURCE_PATH}/LICENSE.leveldb DESTINATION ${CURRENT_PACKAGES_DIR}/share/rocksdb)
```
as
```
file(INSTALL ${SOURCE_PATH}/LICENSE.Apache DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
file(COPY ${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake ${SOURCE_PATH}/LICENSE.leveldb DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/rocksdb/portfile.cmake (hunk 1 above)
Comment (kobykahane) [reply ports/rocksdb/portfile.cmake]: Done.

File: ports/rocksdb/portfile.cmake (hunk 2 above)
Comment (kobykahane) [reply ports/rocksdb/portfile.cmake]: Done.

**Review by NancyLi1013 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Need to test features.

**Comment by NancyLi1013** [comment]
Hi @kobykahane 

It seems that there are some problems with features `zstd `and `snappy`.
The errors like this:
```
CMake Error at F:/10123/vcpkg/scripts/buildsystems/vcpkg.cmake:217 (_add_library):
  Target "rocksdb" links to target "snappy::snappy" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:802 (add_library)

CMake Error at F:/10123/vcpkg/scripts/buildsystems/vcpkg.cmake:217 (_add_library):
  Target "rocksdb" links to target "zstd::zstd" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:802 (add_library)
```
Can these features build successfully on your local? Could you help confirm this?

**Comment by kobykahane** [comment]
> Hi @kobykahane
> 
> It seems that there are some problems with features `zstd `and `snappy`.
> The errors like this:
> 
> ```
> CMake Error at F:/10123/vcpkg/scripts/buildsystems/vcpkg.cmake:217 (_add_library):
>   Target "rocksdb" links to target "snappy::snappy" but the target was not
>   found.  Perhaps a find_package() call is missing for an IMPORTED target, or
>   an ALIAS target is missing?
> Call Stack (most recent call first):
>   CMakeLists.txt:802 (add_library)
> 
> CMake Error at F:/10123/vcpkg/scripts/buildsystems/vcpkg.cmake:217 (_add_library):
>   Target "rocksdb" links to target "zstd::zstd" but the target was not found.
>   Perhaps a find_package() call is missing for an IMPORTED target, or an
>   ALIAS target is missing?
> Call Stack (most recent call first):
>   CMakeLists.txt:802 (add_library)
> ```
> 
> Can these features build successfully on your local? Could you help confirm this?

The issue with snappy appears to be case - if I patch it to `Snappy::snappy` instead it seems to work.
With `zstd` the situation is more complicated. I changed `zstd::zstd` to just `zstd` but then I get a missing `zstd.lib` error during build because the `static` triplets call it `zstd_static.lib` or `zstd_staticd.lib`. I've tried modifying `Findzstd.cmake` by adding these names to the `find_library` invocations there, but that seems to have no effect. Any advice on how to proceed?

**Comment by JackBoosY** [comment]
Using snappy should provide `Snappy::snappy`, zstd does not provide an exported cmake file, and cmake does not provide a `Findzstd.cmake`, so I think you should write a `Findzstd.cmake` based on the zstd-specific file or use `find_path` / `find_library` to find it.

**Comment by NancyLi1013** [comment]
I noticed there is` Findzstd.cmake` in `ports/rocksdb` folder.
As for `zstd`, we also need to add `zstd_static` here since building static, these libs are required.
```
find_library(ZSTD_LIBRARY_DEBUG NAMES zstd zstd_static)
find_library(ZSTD_LIBRARY_RELEASE NAMES zstd zstd_static)
```
BTW, if possible, please also check if this can work on Linux.

In CMakeLists.txt, we might need to upate `zstd::zstd` as `${ZSTD_LIBRARY}`.

As for `snappy`, either `Snappy::snappy` or `${SNAPPY_LIBRARY}` is ok.

**Comment by kobykahane** [comment]
Fixed `snappy` and `zstd` features. Tested build on `x64-windows-static`, `x86-windows-static` and `x64-linux`.

**Review by NancyLi1013 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Need to test features.

**Comment by NancyLi1013** [comment]
All features have been tested pass with the foolowing triplets:
- x86-windows
- x64-windows
- x64-windows-static

**Comment by strega-nil** [comment]
Alright, I'm happy with this change! Thanks @kobykahane :)

**Pull request merged**